src/helpers/routes.js

```diff
@@ -149,7 +150,7 @@ export function makeRoutes (baseRoutes, {
       // - If "router.trailingSlash" is not specified then default to no trailing slash (like Nuxt)
       // - Children with relative paths must not start with slash so don't append if path is empty.
       if (path.length) { // Don't replace empty (child) path with a slash!
-        path = path.replace(/\/+$/, '') + (trailingSlash ? '/' : '') || (isChildWithRelativePath ? '' : '/')
+        path = trailingSlash ? withTrailingSlash(path, true) : withoutTrailingSlash(path, true) || (isChildWithRelativePath ? withoutTrailingSlash(path, true) : withTrailingSlash(path, true))
```

Comment:
The `(isChildWithRelativePath ? '' : '/')` could  stay as before, I think. Even if the new logic is equivalent (hard to say for sure), the old one seems easier to read.

Also shouldn't `trailingSlash ? withTrailingSlash(path, true) : withoutTrailingSlash(path, true)` be wrapped in parenthesis? Otherwise the logic might not work exactly as intended.

Reply:
Good catch, I will add a parenthesis!